HW0/P5/P5.txt

```diff
@@ -1 +1,27 @@
 # Your answers here.
+1. The cashier needs to take number of bags - 1 time to count:
+   256 - 1 = 255 seconds
+
+2. Now the bookstore can add 16 numbers each time, but the count of next round needs to wait the previous round ended:
+   16 + 8 + 4 + 2 + 1 + 1 + 1 + 1 = 34 seconds
+
+3. Now each round only needs 1 second to count:
+   1 + 1 + 1 + 1 + 1 + 1 + 1 + 1 = log2(256) = 8 seconds
+
+4. See P5.png. Single cashier takes N - 1 seconds. Infinite cashiers takes ceiling(log2(N)) seconds.
+   Plot number of bags from 2 to 1200
+
+5. The communicating time takes effect since the second round count. Each round increases 1 second.
+   1 + 2 + 2 + 2 + 2 + 2 + 2 + 2 = 15 seconds
+ 
+6. The cashier who holds all the 256 bags in the beginning distributes the bags to other cashiers. They distribute the bags based on this rule: give the half number of bags in hand to another cashier who has no bags in hand.
```

Comment:
I interpreted this question to mean that the person holding all the bags in the beginning can only distribute them one or two at a time. It looks like your interpretation was different, i.e. the end result of the distribution is that each cashier is only holding one or two bags, but the person initially holding the bags can distribute more than one/two at a time.  Your interpretation is definitely faster (e.g. 25 seconds in part 1, as opposed to 260 seconds in my interpretation)!